dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/redshift/RedshiftClusterUsageMetricsTask.java

```diff
@@ -0,0 +1,196 @@
+                              Stream.of(new Object[] {cluster.getClusterIdentifier()}, metricData)
+                                  .flatMap(Stream::of)
+                                  .toArray());
+                }));
+  }
+
+  private List<Cluster> listClusters() {
+    AmazonRedshift client = redshiftApiClient();
+    return client.describeClusters(new DescribeClustersRequest()).getClusters();
+  }
+
+  private Stream<Pair<Date, Double>> getMetricDataPoints(
+      String clusterId, MetricConfig metricConfig) {
+    AmazonCloudWatch client = cloudWatchApiClient();
+    GetMetricStatisticsRequest request =
+        new GetMetricStatisticsRequest()
+            .withMetricName(metricConfig.name().name())
+            .withStatistics(metricConfig.type().name())
+            .withNamespace(REDSHIFT_NAMESPACE)
+            .withDimensions(new Dimension().withName("ClusterIdentifier").withValue(clusterId))
+            .withStartTime(Date.from(interval.getStartUTC().toInstant()))
+            .withEndTime(Date.from(interval.getEndExclusiveUTC().toInstant()))
+            .withPeriod(metricDataPeriod());
+
+    GetMetricStatisticsResult result = client.getMetricStatistics(request);
+    return result.getDatapoints().stream()
+        .map(
+            datapoint ->
+                Pair.of(datapoint.getTimestamp(), getDatapointValue(metricConfig, datapoint)));
+  }
+
+  private Double getDatapointValue(MetricConfig metricConfig, Datapoint datapoint) {
+    switch (metricConfig.type()) {
+      case Maximum:
+        return datapoint.getMaximum();
+      case Average:
+        return datapoint.getAverage();
+      default:
+        return null;
+    }
+  }
+
+  private Stream<Object[]> zipMetricStreams(List<Stream<Pair<Date, Double>>> metricStreams) {
+    Map<String, Double[]> metricsMap = new HashMap<String, Double[]>();
+
+    for (int i = 0; i < metricStreams.size(); i++) {
+      Integer metricIndex = Integer.valueOf(i);
+      metricStreams
+          .get(metricIndex)
+          .forEach(
+              dataPoint -> {
+                String dateString = DATE_FORMAT.format(dataPoint.getKey().toInstant());
+                if (!metricsMap.containsKey(dateString)) {
+                  metricsMap.put(dateString, new Double[metricStreams.size()]);
+                }
+                metricsMap.get(dateString)[metricIndex] = dataPoint.getValue();
+              });
+    }
+
+    return metricsMap.keySet().stream()
+        .map(
+            metricDate -> {
+              return Stream.of(new Object[] {metricDate}, metricsMap.get(metricDate))
+                  .flatMap(Stream::of)
+                  .toArray();
+            })
+        .sorted((r1, r2) -> ((String) r1[0]).compareTo((String) r2[0]));
+  }
+
+  /**
+   * Returns available metric period based on the interval time.
+   * https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html
+   */
+  private int metricDataPeriod() {
```

Comment:
I would suggest changing the return type to `Duration` and convert to seconds only later, when passing the value to the external API.